### DISCUSSION TITLE
How is E2E encryption handled with Electric SQL?

### DISCUSSION THREAD

**Discussion by fev4** [electric-sql/electric#1080] [Questions]
Hey there,

Totally new to electric-sql. I was wondering, how is E2E encryption handled when it comes to communication between clients and the sync engine?

The context is we are building a budgeting app with a local-first approach on a PWA. We are using Dexie to interface with indexeddb. Pretty recently we have started exploration of syncing mechanisms between clients and electric-sql does sound like a very interesting approach.

But given we want to provide users with a decent level of security and privacy, the question of E2EE comes to mind and then we start to wonder if electric-sql provides any mechanism to make this simpler.

Thanks for any guidance

**Comment by alco** [comment]
Hi @fev4.

We don't have anything ready yet but it is on our roadmap. The tentative design is to provide a hook in the client library that will let you plug in your encryption/decryption algorithms.

We're merging in support for BYTEA/BLOB types soon (see https://github.com/electric-sql/electric/pull/1101), that will unlock first explorations into the realm of E2E encryption.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
